test/langtools/tools/javac/valhalla/value-objects/ValueObjectCompilationTests.java

```diff
@@ -308,6 +308,32 @@ void bar(Object o) {
                     }
                 }
                 """);
+        assertFail("compiler.err.type.found.req",
```

Comment:
Might be good to try the full set of options:
* --source 23, --enable-preview
* --source 23
* --release 23, --enable-preview
* --release 23

(since when release is specified, the code path is different)